services/horizon/internal/app.go

```diff
@@ -88,13 +88,15 @@ func (a *App) GetCoreSettings() actions.CoreSettings {
 	return a.coreSettings.get()
 }
 
+const tickerDuration = 1 * time.Second
```

Comment:
Just a suggestion regarding the name, but I think we should increase timeout to 10 seconds. The aim of this PR is to make sure ticker doesn't hang indefinitely but called `Tick` functions can take more than 10 seconds in certain circumstances. Ex. txsub (without #3563) can take a few sec. when many txs are submitted, possibly reap can be slow with many ledgers to delete.
```suggestion
const maxTickerDuration = 10 * time.Second
```

Reply:
Are you also suggesting we change it to ensure there is only a single `.Tick()` running at a time?

Reply:
Can you elaborate? Before and after this change there can be only one `App.Tick()` running concurrently because of the way `App.run()` works.

Reply:
i.e. Are you suggesting we should attempt to launch a `.Tick()` every second, but abort if one is already running. Or, should we only attempt to launch a `.Tick()` every 10s.

Reply:
We should attempt to lunch it 1s after the previous was done (this is existing behaviour). We should make sure it's not running more than 10s (by sending a context with timeout). The code suggestion from the first comment implements this.

Reply:
> We should attempt to lunch it 1s after the previous was done (this is existing behaviour).

The existing behaviour is that it launches a new one every second (with no regard to the previous).


> We should make sure it's not running more than 10s (by sending a context with timeout). 

👍 


> The code suggestion from the first comment implements this.

Not quite, but sgtm. Will update to do this.